networkx/convert_matrix.py

```diff
@@ -258,18 +283,25 @@ def to_pandas_edgelist(
     source_nodes = [s for s, t, d in edgelist]
     target_nodes = [t for s, t, d in edgelist]
 
-    all_keys = set().union(*(d.keys() for s, t, d in edgelist))
-    if source in all_keys:
+    all_attrs = set().union(*(d.keys() for s, t, d in edgelist))
+    if source in all_attrs:
         raise nx.NetworkXError(f"Source name '{source}' is an edge attr name")
-    if target in all_keys:
+    if target in all_attrs:
         raise nx.NetworkXError(f"Target name '{target}' is an edge attr name")
 
     nan = float("nan")
-    edge_attr = {k: [d.get(k, nan) for s, t, d in edgelist] for k in all_keys}
+    edge_attr = {k: [d.get(k, nan) for s, t, d in edgelist] for k in all_attrs}
```

Comment:
```suggestion
    edge_attr = {k: [d.get(k, nan) for _, _, d in edgelist] for k in all_attrs}
```
I doubt this matters for performance, but may help drive the point home that the goal of the list comp is extracting keys only. Feel free to ignore!